lib/active_merchant/billing/gateways/ixopay.rb

```diff
@@ -130,6 +133,15 @@ def add_debit(xml, money, options)
         end
       end
 
+      def add_refund(xml, money, authorization)
+        xml.refund do
+          xml.transactionId           new_transaction_id
+          xml.referenceTransactionId  authorization.split('|')[0]
```

Comment:
It looks like if a ```nil``` authorization is passed in, we'll generate a ```NoMethodError``` here. It would probably be more helpful for the code to survive to where we make the request to the gateway. That way, we'd ultimately end up with a user-visible error that indicates the authorization (or ```referenceTransactionId```) was invalid. So, a more robust statement would just be null-safe, perhaps ```authorization&.split.try(:[], 0)``` or just a ternary checking ```.present?```.

Reply:
Thanks for pointing this out! I've updated the code as well as both the unit and remote tests to ensure that the code is able to handle `nil` for the authorization.